File: Shooter/Rng.m
//
//  Rng.m
//  Shooter
//
//  Created by Sven A. Schmidt on 03/11/2013.
//  Copyright (c) 2013 feinstruktur. All rights reserved.
//

#import "Rng.h"


@implementation Rng


+ (double)uniform
{
   return (double)arc4random() / UINT32_MAX;
}


+ (double)uniform:(double)max
{
    return [self uniform]*max;
}


+ (double)uniformMin:(double)min max:(double)max
{
    return [self uniform:(max - min)] - min;
}


@end

File: Shooter/Rng.swift
//
//  Rng.swift
//  Shooter
//
//  Created by Sven A. Schmidt on 13/06/2014.
//  Copyright (c) 2014 feinstruktur. All rights reserved.
//

import UIKit


class Rng: NSObject {

    class func uniform() -> Double {
        return Double(arc4random()) / Double(UInt32.max);
    }
    
    
    class func uniform(max: Double) -> Double {
        return uniform() * max
    }
    
    
    class func uniformMin(min: Double, max: Double) -> Double {
        return min + uniform(max - min)
    }
    
    class func uniformRange(range: CGVector) -> Double {
        return uniformMin(range.dx, max: range.dy)
    }

}
